docs/source/contributing/docs/types.rst

```diff
@@ -0,0 +1,134 @@
+like ``direction=(0, 1, 0)``. You could type it as ``InternalPoint3D`` and
+the typechecker and linter would be happy; however, this makes the code harder
+to understand.
+
+As a general rule, if a parameter is called ``direction`` or ``axis``,
+it should be typehinted as some form of ``Vector``.
+
+.. warning::
+
+   This is not always true. For example as of Manim 0.18.0, the direction
+   parameter of the Vector mobject should be ``Point2D | Point3D``
+   as it can also accept ``tuple[float, float]`` and ``tuple[float, float, float]``.
+
+Colors
+------
+The interface manim provides for working with colors is :class:`.ManimColor`.
+The main color types Manim supports are RGB, RGBA, and HSV. You will want
+to typehint a function depending on which type it uses. If any color will work,
+you will need something like:
+
+.. code-block:: python
+
+   if TYPE_CHECKING:
+       from manim.utils.color import ParsableManimColor
+
+   # typehint stuff with ParsableManimColor
+
+
+
+Béziers
+-------
+Manim internally represents a ``Mobject`` by a collection of points. These
+points represent Bézier curves, which are a way of representing a curve using a
+sequence of points.
+
+.. note::
+
+   To learn more about Béziers, take a look at https://pomax.github.io/bezierinfo/
+
+
+Manim supports two different renderers, which each have different representations of
+Béziers: Cairo uses cubic Bézier curves, while OpenGL uses quadratic Bézier curves.
+
+Typehints like ``BezierPoints`` represent a single bezier curve, and ``BezierPath`` is
+essentially a sequence of ``BezierPoints``. A ``Spline`` is when a ``BezierPath``
+forms a closed curve. Manim also provides more specific type aliases when working with
+quadratic or cubic curves, and they are prefixed with their respective type (e.g. ``CubicBezierPoints``,
+which is a ``BezierPoints`` that specifically applies to cubic Bézier curves).
+
+
+Functions
+---------
+Throughout the codebase, many different types of functions are used. The most obvious example
+is a rate function, which takes in a float and outputs a float (``Callable[[float], float]``).
+Another example is for overriding animations. One will often need to map a :class:`.Mobject`
+to an overridden :class:`Animation`, and for that we have the ``FunctionOverride`` typehint.
+
+``PathFuncType`` and ``MappingFunction`` are more niche, but are related to moving objects
+along a path, or applying functions. If you need to use it, you'll know.
+
+
+Images
+------
+There are several representations of images in manim. The most common is
+the representation as a numpy array of floats representing the pixels of an image.
+This is especially common when it comes to the OpenGL renderer.
+
+This is the usecase of the ``Image`` typehint. Sometimes, manim may use ``PIL.Image``,
+in which case one should use that typehint instead.
+Of course if a more specific type of image is needed, it can be annotated as such.
```

Comment:
```suggestion
This is the use case of the :class:`~.Image` type hint. Sometimes, Manim may use ``PIL.Image``,
in which case one should use that type hint instead.
Of course, if a more specific type of image is needed, it can be annotated as such.
```